### ISSUE TITLE
name filter for images does not work if the name contains a host with a port

### ISSUE THREAD

**Issue by morxa** [docker/docker-py#908]
If the repository name contains a host with a port, then filtering for that repository name does not work. As an example, if you have an image `myregistry:5000/test`, then `client.images(name='myregistry:5000/test')` returns an empty list.

### How to reproduce

1. Tag an existing image to a new name with a port in the registry address:

        $ docker tag ubuntu myregistry:5000/test
1. Use `docker-py` to check if the image exists:

        import docker
        docker_client = docker.Client(base_url='unix:///var/run/docker.sock')
        docker_client.images(name='myregistry:5000/test')

*expected result:* The returned list should contain the newly tagged image.
*actual result:* `[]`

### versions
docker-py 1.6.0
Python 2.7.6
Docker version 1.9.1-cs3, build 28d463f

**Comment by shin-** [comment]
Works for me?
```python
In [1]: import docker

In [2]: c= docker.Client()

In [3]: c.images(name='registry.com:2222/hello')
Out[3]: 
[{u'Created': 1452128968,
  u'Id': u'sha256:9c4901f3703a178bee19d02a0dd22794a93c6658ccbc53af44bada7e3703edc6',
  u'Labels': None,
  u'ParentId': u'',
  u'RepoDigests': None,
  u'RepoTags': [u'registry.com:2222/hello:latest'],
  u'Size': 84894442,
  u'VirtualSize': 84894442}]

In [4]: docker.version
Out[4]: '1.6.0'
```

**Comment by morxa** [comment]
```
% docker images | grep hello
hello                                                latest              6b40fe7724bd        11 weeks ago        15.6 MB
registry.com:2222/hello                              latest              6b40fe7724bd        11 weeks ago        15.6 MB
registry.com/hello                                   latest              6b40fe7724bd        11 weeks ago        15.6 MB
```


```
In [1]: import docker

In [2]: docker_client = docker.Client()

In [3]: docker_client.images(name='registry.com:2222/hello')
Out[3]: []

In [4]: docker_client.images(name='registry.com/hello')
Out[4]: 
[{u'Created': 1446581252,
  u'Id': u'6b40fe7724bd29107f6182ca2befec011cdf524b23ebc4c9a33591d6b7aea4ee',
  u'Labels': None,
  u'ParentId': u'2b4c55187a27a43f6c5aebab707e75cffee14f7c3cc02e4f74429992bd5f7db2',
  u'RepoDigests': [],
  u'RepoTags': [u'registry.com/hello:latest'],
  u'Size': 0,
  u'VirtualSize': 15602184}]

In [5]: docker.version
Out[5]: '1.6.0'

```

**Comment by shin-** [comment]
As it turns out, this is a Docker engine bug that will be fixed in the upcoming 1.10 release. See https://github.com/docker/docker/issues/18181 ;
I couldn't reproduce the issue because I'm currently using the 1.10 RC.
